sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureSeekableByteChannel.java

```diff
@@ -0,0 +1,191 @@
+
+    private final NioBlobInputStream inputStream; // Always set the mark at 0 w/ max int. repositioning is a reset and then a skip. Validate length
+    private final NioBlobOutputStream outputStream;
+    private long position; // Needs to be threadsafe?
+    private boolean closed = false;
+    private final Path path;
+
+    AzureSeekableByteChannel(NioBlobInputStream inputStream, Path path) {
+        this.inputStream = inputStream;
+        inputStream.mark(Integer.MAX_VALUE);
+        this.outputStream = null;
+        this.position = 0;
+        this.path = path;
+    }
+
+    AzureSeekableByteChannel(NioBlobOutputStream outputStream, Path path) {
+        this.outputStream = outputStream;
+        this.inputStream = null;
+        this.position = 0;
+        this.path = path;
+    }
+
+    @Override
+    public int read(ByteBuffer dst) throws IOException {
+        AzurePath.ensureFileSystemOpen(this.path);
+        validateOpen();
+        validateReadMode();
+
+        // See comments in position()
+        if (this.position > this.size()) {
+            return -1;
+        }
+
+        int count = 0;
+
+        int len = dst.remaining();
+        byte[] buf = new byte[len];
+
+        while (count < len) {
+            int retCount = this.inputStream.read(buf, count, len - count);
+            if (retCount == -1) {
+                break;
+            }
+            count += retCount;
+        }
+        dst.put(buf, 0, count);
+        this.position += count;
+
+        return count;
+        // Would it be easier to just do the downloads directly?
+    }
+
+    @Override
+    public int write(ByteBuffer src) throws IOException {
+        AzurePath.ensureFileSystemOpen(this.path);
+        validateOpen();
+        validateWriteMode();
+
+        int length = src.remaining();
+
+        this.position += src.remaining();
+        byte[] buf = new byte[length];
+        src.get(buf);
+        this.outputStream.write(buf);
+
+        return length;
+    }
+
+    @Override
+    public long position() throws IOException {
+        AzurePath.ensureFileSystemOpen(this.path);
+        validateOpen();
+
+        return this.position;
+    }
+
+    @Override
+    public AzureSeekableByteChannel position(long newPosition) throws IOException {
+        AzurePath.ensureFileSystemOpen(this.path);
+        validateOpen();
+        validateReadMode();
+
+        if (newPosition < 0) {
+            throw LoggingUtility.logError(logger, new IllegalArgumentException("Seek position cannot be negative"));
+        }
+
+        /*
+        The javadoc says seeking past the end for reading is legal and that it should indicate the end of the file on
+        the next read. StorageInputStream doesn't allow this, but we can get around that by just playing with the
+        position variable and skipping the actual read. We'll check in read if we've seeked past the end and short
```

Comment:
Can you clarify this doc a little - "we can get around that by modifying the position variable and skipping the actual read (when read is called next). "